mobileChatExamples/connectReactNativeChat/local-proxy-backend/server.js

```diff
@@ -0,0 +1,107 @@
+const express = require('express')
+const cors = require('cors')
+const { ConnectClient, StartChatContactCommand } = require('@aws-sdk/client-connect')
+
+const app = express();
+const PORT = 9000;
+
+const validateEnvVars = () => {
+  const requiredVars = [
+    'AMAZON_CONNECT_INSTANCE_ID',
+    'AMAZON_CONNECT_CONTACT_FLOW_ID',
+    'AMAZON_CONNECT_REGION',
+    'AWS_ACCESS_KEY_ID',
+    'AWS_SECRET_ACCESS_KEY',
+    'AWS_SESSION_TOKEN'
+  ];
+
+  const missingVars = requiredVars.filter(varName => !process.env[varName]);
+
+  if (missingVars.length > 0) {
+    console.error(`Missing required environment variables: ${missingVars.join(', ')}`);
+    process.exit(1);
+  }
+};
+
+validateEnvVars();
+
+const CONFIG = {
+  instanceId: process.env.AMAZON_CONNECT_INSTANCE_ID,
+  contactFlowId: process.env.AMAZON_CONNECT_CONTACT_FLOW_ID,
+  region: process.env.AMAZON_CONNECT_REGION || 'us-west-2',
+};
+const CREDENTIALS = {
+  accessKeyId: process.env.AWS_ACCESS_KEY_ID,
+  secretAccessKey: process.env.AWS_SECRET_ACCESS_KEY,
+  sessionToken: process.env.AWS_SESSION_TOKEN,
+};
+
+app.use(express.json());
+app.use(cors());
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

It appears that when `Access-Control-Allow-Origin` is specified as `*` wildcard, it allows access to all unwanted origins. To prevent unlimited access and a possible cross-site request forgery (CSRF) attack, limit Access-Control-Allow-Origin to only trusted origins.[Learn more](https://cwe.mitre.org/data/definitions/942.html)